core-jdk8/src/main/java/org/mapstruct/ValueMapping.java

```diff
@@ -0,0 +1,125 @@
+ */
+package org.mapstruct;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Configures the mapping of source constant value to target constant value.
+ * <p>
+ * Supported mappings are
+ * <ol>
+ * <li>Enumeration to Enumeration</li>
+ * </ol>
+ * <p>
+ * <B>Example 1:</B>
+ * <pre>
+ * <code>
+ * public enum OrderType { RETAIL, B2B, EXTRA, STANDARD, NORMAL }
+ *
+ * public enum ExternalOrderType { RETAIL, B2B, SPECIAL, DEFAULT }
+ *
+ * &#64;ValueMappings({
+ *    &#64;ValueMapping(source = "EXTRA", target = "SPECIAL"),
+ *    &#64;ValueMapping(source = "STANDARD", target = "DEFAULT"),
+ *    &#64;ValueMapping(source = "NORMAL", target = "DEFAULT")
+ * })
+ * ExternalOrderType orderTypeToExternalOrderType(OrderType orderType);
+ * </code>
+ * Mapping result:
+ * ╔═════════════════════╦════════════════════════════╗
+ * ║ OrderType           ║ ExternalOrderType          ║
+ * ╠═════════════════════╬════════════════════════════╣
+ * ║ &lt;NULL&gt;              ║ &lt;NULL&gt;                     ║
+ * ║ OrderType.EXTRA     ║ ExternalOrderType.SPECIAL  ║
+ * ║ OrderType.STANDARD  ║ ExternalOrderType.DEFAULT  ║
+ * ║ OrderType.NORMAL    ║ ExternalOrderType.DEFAULT  ║
+ * ║ OrderType.RETAIL    ║ ExternalOrderType.RETAIL   ║
+ * ║ OrderType.B2B       ║ ExternalOrderType.B2B      ║
+ * ╚═════════════════════╩════════════════════════════╝
+ * </pre>
+ * MapStruct will <B>WARN</B> on incomplete mappings. However, if for some reason no match is found an
+ * {@link java.lang.IllegalStateException} will be thrown.
+ * <p>
+ * <B>Example 2:</B>
+ * <pre>
+ * <code>
+ * &#64;ValueMappings({
+ *    &#64;ValueMapping( source = "&lt;NULL&gt;", target = "DEFAULT" ),
+ *    &#64;ValueMapping( source = "STANDARD", target = "&lt;NULL&gt;" ),
+ *    &#64;ValueMapping( source = "&lt;ANY&gt;", target = "SPECIAL" )
+ * })
+ * ExternalOrderType orderTypeToExternalOrderType(OrderType orderType);
+ * </code>
+ * Mapping result:
+ * ╔═════════════════════╦════════════════════════════╗
+ * ║ OrderType           ║ ExternalOrderType          ║
+ * ╠═════════════════════╬════════════════════════════╣
+ * ║ &lt;NULL&gt;              ║ ExternalOrderType.DEFAULT  ║
+ * ║ OrderType.STANDARD  ║ &lt;NULL&gt;                     ║
+ * ║ OrderType.RETAIL    ║ ExternalOrderType.RETAIL   ║
+ * ║ OrderType.B2B       ║ ExternalOrderType.B2B      ║
+ * ║ &lt;ANY&gt;               ║ ExternalOrderType.SPECIAL  ║
+ * ╚═════════════════════╩════════════════════════════╝
+ * </pre>
+ *
+ * @author Sjaak Derksen
+ */
+@Repeatable(ValueMappings.class)
+@Retention(RetentionPolicy.CLASS)
+@Target(ElementType.METHOD)
+public @interface ValueMapping {
+    /**
+     * The source value constant to use for this mapping.
+     *
+     * <p>
+     * <b>Valid values:</b>
+     * <ol>
+     * <li>ENUM IDENTIFIER</li>
```

Comment:
I actually got it from the java spec (see https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html#jls-8.9.1), where the actual name is referred to as 'the identifier'  (and with access modifier + constructor argument is called enum constant. However, it does not any clarity. i'll change..